### DISCUSSION TITLE
Behaviour of the service "Forcible Charge (to a SoC level)" ?

### DISCUSSION THREAD

**Discussion by BenoitVH** [wlcrs/huawei_solar#553] [Setting up the integration]
Hi, I'm really enjoying this integration. Kudo's to the authors(s)

After playing around with some nice dashboards, I would now like to control the charging behaviour of my battery. 
During winter, my solar production is not sufficient and so I would charge my 5KWh battery during the night (=cheaper) for discharge during the day. I also own an electric waterboiler (1.8KW) that I charge at night in winter (during day in summer) This one is controled via a Tuya switch.
I have 3 automations running:
1) start charging my battery to 90% SOC at 1.5KW at 11:30 PM (using "Forcible Charge to a SoC level")
2) power on boiler at 3:15 AM
3) Stop both boiler and battery charge (using "Stop Forcible Charge") at 5:55 AM  (as electricity price increases at 6AM)

**Now here comes the question.**
Resulting behaviour: 
1) Battery starts charging from the grid and reaches the 90 % around 3AM
2) Boiler heats up but using the power from the battery **not from the grid**
3) Both boiler and battery charging stop ; Battery is empty
(see picture at the bottom)

I was hoping that the service "Forcible Charge to a SoC level" would keep the battery charged at SOC level and the boiler would charge from the grid (iso battery) as long as forcible_stop is not called.
I know I can reverse the order to solve my particular problem but this question is really about understanding the forcible service call behaviour as the documentation is not fully explicit about this: https://github.com/wlcrs/huawei_solar/wiki/Force-charge-discharge-battery 

![Screenshot 2023-12-08 152833](https://github.com/wlcrs/huawei_solar/assets/71166296/467649e9-a6d6-48e3-9228-db9e29dc23d2)

**Comment by HolgerMiara** [comment]
I have exactly the same question, even though I don't have a boiler but an electric car that I want to charge during night. 
Now during winter, I charge my electric car at night the when price of electricity is low. I use Jonas Karlsson's [EV Smart Charging integration](https://github.com/jonasbkarlsson/ev_smart_charging) to find the cheapest timespan and to start the charging. 
I omitted draining the battery by defining TOU timespans in the Fusion Solar App. In the period when you expect charging of your car (or heating of your boiler) nothing should be defined (except charging, I guess). 

Looking at it again, it might be smarter to use the service "Huawei Solar: Set Fixed Charge Periods" instestead of "Huawei Solar: Set TOU Periods". I might give it a try it in the future. Pls. keep me updated how you solve it!
